src/workerd/io/worker.c++

```diff
@@ -1438,6 +1440,16 @@ Worker::~Worker() noexcept(false) {
   lock->push(kj::mv(impl));
 }
 
+void Worker::logBotManagementUse() const {
+  // TODO(maybe): If logging this for every worker proves to be too noisy, then we
+  // can figure out a way to log of for workers that access when they do not have
+  // a subscription.
+  if (!impl->loggedBotManagementAccess) {
+    impl->loggedBotManagementAccess = true;
```

Comment:
If a worker is evicted and then recreated, this flag will be reset and we'll log again. Low-traffic workers are normally evicted and recreated on every request, so this could amplify the noise considerably.

Perhaps we should maintain a global HashSet of zones for which we've already logged?

Zone IDs don't exist in workerd, of course, so we'll need some sort of a hook here. Maybe `logBotManagement` should be a method on `RequestObserver` (the metrics object) instead of `Worker`? That makes some logical sense and also makes it easy to access internal information in the implementation.

Reply:
Yeah, good point. We really do not need the logging at all in workerd so I think it's fine if it ends up being a non-op there. Moving to `RequestObserver` makes sense.